docker/web-and-data/Dockerfile

```diff
@@ -0,0 +1,39 @@
+# This image includes both the web app and the data import related scripts in
+# one image. Because of historic reasons some of these scripts run java
+# artifacts under the hood. It therefore currently makes more sense to have a
+# single image supporting both instead of moving the scripts out to a separate
+# container.
+FROM maven:3.5.4 as build
+COPY $PWD /cbioportal
+WORKDIR /cbioportal
+RUN mvn -DskipTests clean install
+
+FROM shipilev/openjdk-shenandoah:8
+RUN mkdir -p /cbioportal
+COPY --from=build /cbioportal/portal/target/cbioportal*.war /app.war
+COPY --from=build /cbioportal/portal/target/dependency/webapp-runner.jar /webapp-runner.jar
+# copy over core jar and scripts
+COPY --from=build /cbioportal/core /cbioportal/core
+COPY --from=build /cbioportal/scripts /cbioportal/scripts
+COPY --from=build /cbioportal/requirements.txt /cbioportal/requirements.txt
+
+# install build and runtime dependencies
+# ignore update failures
+RUN apt-get update; apt-get install -y --no-install-recommends \
+        build-essential \
+        libmysqlclient-dev \
+        python3 \
+        python-dev \
+        python3-dev \
+        python3-pip \
+	&& rm -rf /var/lib/apt/lists/* \
+    && pip3 install -r /cbioportal/requirements.txt
+
+# add importer scripts to PATH for easy running in containers
+RUN find /cbioportal/core/src/main/scripts/ -type f -executable \! -name '*.pl'  -print0 | xargs -0 -- ln -st /usr/local/bin
+
+# put config files in this folder if you want to override config
+ENV PORTAL_HOME=/cbioportal
+
+# TODO: fix the workdir-dependent references to '../scripts/env.pl' and do this:
+# RUN find $PWD/core/src/main/scripts/ -type f -executable \! \( -name env.pl -o -name envSimple.pl \)  -print0 | xargs -0 -- ln -st /usr/local/bin
```

Comment:
@inodb , is it possible to remove <none> image after docker build, since it was 3.28GB from my test?

Reply:
@khzhu, the way I understand it, Docker will keep the intermediate build container only as a cache. It would be used if you download an updated version of the runtime base image (containing the Debian base system and OpenJDK runtime), and you want to build an image that combines the cBioPortal code you already compiled with the new environment. It should not be needed at buildtime, so if you don't mind having to recompile cBioPortal in such a case, you could safely remove it after the build step.

Reply:
@khzhu, does that work for you? Do you feel that removing images should be explicitly mentioned in the documentation?

Reply:
I would suggest to document those "none" images, so users know where those images come from. You might also want to include a docker command to remove dangled images.

Reply:
- why do we need python-dev and python3-dev (line 26, 27)? 
- any reasons to use mvn 3.5.4?

Reply:
@khzhu @fedde-s The idea of the documentation updates for this PR is to get people to run the portal using docker images. We would provide these images on docker hub.

I think building your own images and changing them is another use case for developers. Ideally people deploying the portal wouldn't have to do this.

@fedde-s has written a substantial amount of documentation on how to use Docker as a developer. But I think to include it, it would need some restructuring so we can clearly separate these use cases. The `none` images point can be addressed in that documentation

Reply:
Sounds good to me! If @khzhu agrees, can one of you phrase an issue report to track the follow-up task of refining the docs for that use-case for devs new to Docker?

Reply:
created this one to track https://github.com/cBioPortal/cbioportal/issues/6131